eslint-plugin/rules/validate-imports.js

```diff
@@ -16,13 +16,37 @@ module.exports = {
       badFragmentSpread:
         'Fragment "{{ fragment }}" cannot be spread here as objects of type "{{ objectType }}" can never be of type "{{ fragmentType }}"',
       unusedFragmentDefinition: 'Fragment "{{ fragmentName }}" is never used',
+      fileNotFound: 'no such file: "{{ importIdentifier }}" starting at: "{{ basedir }}"',
     },
     docs: {
       description: 'ensure imports are valid',
       category: 'problem',
     },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          importResolver: {
+            type: 'string',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
   },
   create(context) {
+    let resolveImport;
+
+    let importResolver =
+      Array.isArray(context.options) &&
+      context.options.length > 0 &&
+      context.options[0].importResolver;
+    if (typeof importResolver === 'string') {
+      resolveImport = require(importResolver);
```

Comment:
Dynamic `require` like this has some problems, as it is vulnerable to node_module topology changes, yarn/npm linking etc.  In general, unless we can explicity depend on something, or we can provide the appropriate basedir, we should only allow importing of absolute paths.